registration-processor/init/registration-processor-packet-receiver-stage/src/main/java/io/mosip/registration/processor/packet/receiver/config/PacketReceiverConfig.java

```diff
@@ -1,120 +1,168 @@
+import io.mosip.kernel.core.virusscanner.spi.VirusScanner;
+import io.mosip.kernel.idvalidator.rid.impl.RidValidatorImpl;
+import io.mosip.registration.processor.core.abstractverticle.MessageDTO;
+import io.mosip.registration.processor.core.spi.filesystem.manager.FileManager;
+import io.mosip.registration.processor.packet.manager.decryptor.Decryptor;
+import io.mosip.registration.processor.packet.manager.decryptor.DecryptorImpl;
+import io.mosip.registration.processor.packet.manager.dto.DirectoryPathDto;
+import io.mosip.registration.processor.packet.manager.idreposervice.IdRepoService;
+import io.mosip.registration.processor.packet.manager.idreposervice.impl.IdRepoServiceImpl;
+import io.mosip.registration.processor.packet.manager.service.impl.FileManagerImpl;
+import io.mosip.registration.processor.packet.receiver.builder.PacketReceiverResponseBuilder;
+import io.mosip.registration.processor.packet.receiver.exception.handler.PacketReceiverExceptionHandler;
+import io.mosip.registration.processor.packet.receiver.service.PacketReceiverService;
+import io.mosip.registration.processor.packet.receiver.service.impl.PacketReceiverServiceImpl;
+import org.springframework.core.env.Environment;
+
+/**
+ * The Class PacketReceiverConfig.
+ */
+/**
+ * @author Mukul Puspam
+ *
+ */
+@Configuration
+@EnableAspectJAutoProxy
+public class PacketReceiverConfig {
+
+	public static final String VIRUS_SCANNER_PROVIDER = "mosip.regproc.virusscanner.provider";
+
+	private static Logger logger = RegProcessorLogger.getLogger(PacketReceiverConfig.class);
+
+	@Autowired
+	private Environment env;
+
+	/**
+	 * PacketReceiverService bean.
+	 *
+	 * @return the packet receiver service
+	 */
+	@Bean
+	public PacketReceiverService<File, MessageDTO> getPacketReceiverService() {
+		return new PacketReceiverServiceImpl();
+	}
+
+	/**
+	 * GlobalExceptionHandler bean.
+	 *
+	 * @return the global exception handler
+	 */
+	@Bean
+	public PacketReceiverExceptionHandler getGlobalExceptionHandler() {
+		return new PacketReceiverExceptionHandler();
+	}
+
+	/**
+	 * Gets the packet receiver response builder.
+	 *
+	 * @return the packet receiver response builder
+	 */
+	@Bean
+	public PacketReceiverResponseBuilder getPacketReceiverResponseBuilder() {
+		return new PacketReceiverResponseBuilder();
+	}
+
+	/**
+	 * Virus scanner service. Load virus scanner during runtime from property mosip.regproc.virusscanner.provider
+	 *
+	 * @return the virus scanner
+	 */
+	@Bean
+	@Lazy
+	public VirusScanner<Boolean, InputStream> virusScannerService() throws ClassNotFoundException, IllegalAccessException, InstantiationException {
+		if (StringUtils.isNotBlank(env.getProperty(VIRUS_SCANNER_PROVIDER))) {
+			logger.debug("mosip.regproc.virusscanner.provider is set as ", env.getProperty(VIRUS_SCANNER_PROVIDER),
+					"loading VirusScanner", "");
+			return (VirusScanner) Class.forName(env.getProperty(VIRUS_SCANNER_PROVIDER)).newInstance();
+		} else {
```

Comment:
you don't need the else here. Just a direct log.error would be fine as the if statement any way returns back.